systemverilog-plugin/UhdmAst.cc

```diff
@@ -806,7 +805,19 @@ static AST::AstNode *expand_dot(const AST::AstNode *current_struct, const AST::A
         std::swap(right, sub_dot->children[1]);
         delete sub_dot;
     }
-    if (struct_range) {
+
+    for (size_t i = 0; i < struct_ranges.size(); i++) {
+        auto *struct_range = struct_ranges[i];
```

Comment:
```suggestion
        const auto * const struct_range = struct_ranges[i];
```
(quick look says it's possible, but please verify)

Reply:
LGTM, I've applied this.